### DISCUSSION TITLE
tutorials/lift-off-part3/07-building-the-track-page

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#66] [course feedback]
# tutorials/lift-off-part3/07-building-the-track-page

https://www.apollographql.com/tutorials/lift-off-part3/07-building-the-track-page

**Comment by neogeek13** [comment]
Would be helpful to include `) just before the last ; in the following code sample:

export const GET_TRACK = gql(`
// our query goes here
;

Doing so means that dropping in the next block of code replacing the comment works without debugging (even if it's just a few seconds).

**Reply by mabuyo** [reply]
Good catch! Will update shortly :pray:

**Comment by busanmaz** [comment]
After I've run the codegen, started to get a type error in the `tracks.tsx` component.
We pass the `track` data to the TrackCard component and the type of this prop is `Track` which has a non-optional `modules` field.
But we do not fetch the modules in the `tracksForHome` query, this is causing the following error.
This error is handled if we change the `track` type in the `track-card` component to `Omit<Track, "modules">`.
Is there any better way of handling this error?

Type error:
```
Property 'modules' is missing in type '{ __typename?: "Track" | undefined; id: string; title: string; thumbnail?: string | null | undefined; length?: number | null | undefined; modulesCount?: number | null | undefined; author: { ...; }; }' but required in type 'Track'.ts(2741)

graphql.ts(63, 3): 'modules' is declared here.

track-card.tsx(12, 29): The expected type comes from property 'track' which is declared here on type 'IntrinsicAttributes & { track: Track; }'
```

Type error can be handled as below:
```
const TrackCard: React.FC<{ track: Omit<Track, "modules"> }> = ({track}) => ....
```

**Reply by lizhennessy** [reply]
Hi there @busanmaz ! I'm sorry to hear you're running into this error. I've just gone through the course myself and managed to get things running without seeing it; do you happen to have your code pushed up somewhere that I can take a look? The way we've configured our server's codegen should make it so that we don't need to explicitly omit the "modules" field.

**Comment by Souravsinha96** [comment]
After i have run the codegen, i also got the same error

Property 'modules' is missing in type '{ __typename?: "Track" | undefined; id: string; title: string; thumbnail?: string | null | undefined; length?: number | null | undefined; modulesCount?: number | null | undefined; author: { ...; }; }' but required in type 'Track'.ts

I think there is one issue  while creating schema : we define module as a non nullable, means it has to be there
  type Track {
    id: ID!
    title: String!
    author: Author!
    thumbnail: String
    length: Int
    modulesCount: Int
    "The track's complete description, can be in Markdown format"
    description: String
    "The number of times a track has been viewed"
    numberOfViews: Int
    modules: [Module!]!
  }

So we  need to omit in Track Card componnet in my opiniion.
Please help!!!

**Reply by lizhennessy** [reply]
Hi @Souravsinha96 ! Happy to help. Could you share with me the link to your repo? Or perhaps just paste the contents of the following files from your `server` directory:

- `codegen.ts`
- `models.ts`
- `context.ts`

After you run `npm run generate` on the server side, it would be useful to see the generated `types.ts` file as well!

Thanks so much, I'm looking forward to your response.

Liz

**Reply by Souravsinha96** [reply]
Hi @lizhennessy.,
Link for the repo: https://github.com/Souravsinha96/apollo-odyssey

**Reply by jkpchang** [reply]
@lizhennessy, 

I know why. I ran into this problem too.. that's because we continue using the repo from Part 2 (odyssey-lift-off-part2)  instead of cloning the new repo from Part 3 (odyssey-lift-off-part3).

In Part 2:
```
const TrackCard: React.FC<{ track: Track }> = ({ track }) => {
```

In Part 3:
```
const TrackCard: React.FC<{ track: Omit<Track, "modules"> }> = ({ track }) => {
```

but the tutorial didn't tell us to make that update.
